### PR TITLE
Add create-network cli command

### PR DESCRIPTION
**Pull request by m3dwards**
Look @pinheadmz, a test!

### CODE REVIEW

File: src/warnet/graph.py [hunk 1]
```diff
@@ -17,3 +24,159 @@ def import_json(infile: Path, outfile: Path, cb: str, ln_image: str):
...
+        int(net_answers["nodes"]),
+        int(net_answers["connections"]),
+        net_answers["version"],
+        custom_network_path,
+    )
+    return custom_network_path
+
+
+@click.command()
+def create_network():
```
Comment (josibake) [inline src/warnet/graph.py]: nice! @willcl-ark had suggested previously we use `warnet init` and `warnet new` for initialising or creating new project directories. this matches cargo's new/init. 

if we did that, we could just use `warnet create` for creating a new network? I prefer the smaller cli command names and I don't envision a user creating anything _but_ a new network.

we could also just have one command for initialising, e.g. `warnet init` and use `warnet create` for creating a network, which would be easy enough to do in this PR, and then we can add `warnet new` later if we feel its needed?

File: src/warnet/graph.py [hunk 2]
```diff
@@ -17,3 +24,159 @@ def import_json(infile: Path, outfile: Path, cb: str, ln_image: str):
...
+
+        nodes.append(node)
+
+    network_yaml_data = {"nodes": nodes}
+
+    with open(os.path.join(datadir, "network.yaml"), "w") as f:
+        yaml.dump(network_yaml_data, f, default_flow_style=False)
+
+    # Generate defaults.yaml
+    default_yaml_path = files("resources.networks").joinpath("6_node_bitcoin/node-defaults.yaml")
```
Comment (josibake) [inline src/warnet/graph.py]: note to self: we should move all the hardcoded values into something like config.py or constants.py , so these can be something like `default_yaml_path = config.DEFAULT_YAML_PATH` etc

**Review by josibake — COMMENTED** [review]
code review ACK, looks good. I do think its worth changing `warnet create` to mean `create a network` in this PR, and either only having `warnet init` or `warnet new` and `init` for starting a project.

File: src/warnet/graph.py (hunk 1 above)
Comment (m3dwards) [reply src/warnet/graph.py]: What do you think about dropping create / new?

Comment (josibake) [reply src/warnet/graph.py]: id be fine to only have `warnet init`. initially I was thinking `warnet create` would be helpful when you want to manage multiple warnet directories, but starting to feel like no one uses it and it was premature optimisation.

id much rather we have `warnet init` and then have `warnet create` for creating networks.

**Comment by pinheadmz** [comment]
not sure what to make of this....?


```
(.venv) --> test/graph_test.py
2024-09-04 11:09:59 | INFO    | test     | Logging started
2024-09-04 11:09:59 | INFO    | test     | Warnet test base initialized
Traceback (most recent call last):
  File "/Users/matthewzipkin/Desktop/work/warnet/test/graph_test.py", line 44, in <module>
    test.run_test()
  File "/Users/matthewzipkin/Desktop/work/warnet/test/graph_test.py", line 20, in run_test
    self.directory_exists()
  File "/Users/matthewzipkin/Desktop/work/warnet/test/graph_test.py", line 39, in directory_exists
    self.sut.expect("successfully", timeout=5)
  File "/Users/matthewzipkin/Desktop/work/warnet/.venv/lib/python3.12/site-packages/pexpect/spawnbase.py", line 354, in expect
    return self.expect_list(compiled_pattern_list,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthewzipkin/Desktop/work/warnet/.venv/lib/python3.12/site-packages/pexpect/spawnbase.py", line 383, in expect_list
    return exp.expect_loop(timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthewzipkin/Desktop/work/warnet/.venv/lib/python3.12/site-packages/pexpect/expect.py", line 181, in expect_loop
    return self.timeout(e)
           ^^^^^^^^^^^^^^^
  File "/Users/matthewzipkin/Desktop/work/warnet/.venv/lib/python3.12/site-packages/pexpect/expect.py", line 144, in timeout
    raise exc
pexpect.exceptions.TIMEOUT: Timeout exceeded.
<pexpect.pty_spawn.spawn object at 0x102e76600>
command: /Users/matthewzipkin/Desktop/work/warnet/.venv/bin/warnet
args: ['/Users/matthewzipkin/Desktop/work/warnet/.venv/bin/warnet', 'create-network']
buffer (last 100 chars): b'[m\x0f  26.0\x1b[m\x0f\x1b[K\r\n \x1b[m\x0f  25.1\x1b[m\x0f\x1b[K\r\n \x1b[m\x0f  24.2\x1b[m\x0f\x1b[K\r\n \x1b[m\x0f  23.2\x1b[m\x0f\x1b[K\r\n \x1b[m\x0f  22.2\x1b[m\x0f\x1b[K\r\n\r\n'
before (last 100 chars): b'[m\x0f  26.0\x1b[m\x0f\x1b[K\r\n \x1b[m\x0f  25.1\x1b[m\x0f\x1b[K\r\n \x1b[m\x0f  24.2\x1b[m\x0f\x1b[K\r\n \x1b[m\x0f  23.2\x1b[m\x0f\x1b[K\r\n \x1b[m\x0f  22.2\x1b[m\x0f\x1b[K\r\n\r\n'
after: <class 'pexpect.exceptions.TIMEOUT'>
match: None
match_index: None
exitstatus: None
flag_eof: False
pid: 14835
child_fd: 8
closed: False
timeout: 30
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: None
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0.05
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile(b'successfully')
2024-09-04 11:10:05 | INFO    | test     | Stopping network
2024-09-04 11:10:05 | DEBUG   | test     | Executing warnet command: down
```

**Comment by m3dwards** [comment]
> not sure what to make of this

hmm, not sure, could be that the timeout needs to be longer. I assumed it was seconds but perhaps it's milliseconds.

**Comment by m3dwards** [comment]
Care to give it another go @pinheadmz ?

File: src/warnet/graph.py (hunk 1 above)
Comment (m3dwards) [reply src/warnet/graph.py]: Addressed in [df2856d](https://github.com/bitcoin-dev-project/warnet/pull/524/commits/df2856deeccb24f68471edc02f02ba415236bd8a)

**Review by josibake — APPROVED** [review]
code review ACK, thanks for taking the naming suggestion for the cli command

File: src/warnet/graph.py [hunk 3]
```diff
@@ -17,3 +24,188 @@ def import_json(infile: Path, outfile: Path, cb: str, ln_image: str):
...
+
+@click.command()
+def create():
+    """Create a new warnet network"""
+    try:
+        project_path = Path(os.getcwd())
+        # Check if the project has a networks directory
+        if not (project_path / "networks").exists():
+            click.secho(
+                "The current directory does not have a 'networks' directory. Please run 'warnet init' or 'warnet create' first.",
```
Comment (josibake) [inline src/warnet/graph.py:195]: need to update this help text now that we only have `warnet init`

**Comment by pinheadmz** [comment]
Test passed for me locally now, but CI doesnt look happy even though its green:
https://github.com/bitcoin-dev-project/warnet/actions/runs/10704964245/job/29679197963#step:5:78

Quick guess is test_base tries to call "warnet down" even though it was never up

**Comment by pinheadmz** [comment]
ACK, modulo weird test down thing. can be fixed in follow up

**Comment by josibake** [comment]
> ACK, modulo weird test down thing. can be fixed in follow up

was also running into this on a diff branch, iirc. I think we just need to fix warnet down so that if you call down and there's nothing there, it just gracefully exits (assuming that's actually the issue here)

File: src/warnet/graph.py (hunk 1 above)
Comment (willcl-ark) [reply src/warnet/graph.py]: `cargo` has `cargo init` and `cargo new` which IMO make sense. Both make a new project, one in current dir, and one in the dir provided as arg.

No strong opinion though

**Comment by 256dummy** [comment]
The following sections might be updated with supplementary metadata relevant to reviewers and maintainers.


### Conflicts
Reviewers, this pull request conflicts with the following ones:

* [#531](https://github.com/bitcoin-dev-project/warnet/pull/531) (unified HTML landing page (NOT a dashboard!) by willcl-ark)
* [#509](https://github.com/bitcoin-dev-project/warnet/pull/509) (Update quickstart: minikube and docker-desktop by mplsgrant)
* [#506](https://github.com/bitcoin-dev-project/warnet/pull/506) (Run any executable file on a commander by willcl-ark)

If you consider this pull request important, please also help to review the conflicting pull requests. Ideally, start with the one that should be merged first.

**Comment by josibake** [comment]
would be nice to get this in, can you rebase @m3dwards ? otherwise I can take a crack at resolving the merge conflicts

**Comment by m3dwards** [comment]
Rebased after #536

**Pull request merged**